DependencyInjection/Configurator.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SwiftmailerBundle\DependencyInjection;
+
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * Service configurator.
+ */
+class Configurator
+{
+    /**
+     * @var string
+     */
+    protected $localDomain;
+
+    /**
+     * @var RequestContext
+     */
+    protected $requestContext;
+
+    /**
+     * Sets the local domain based on the current request context.
+     *
+     * @param string         $localDomain    Fallback value if there is no request context.
+     * @param RequestContext $requestContext
+     */
+    public function __construct($localDomain, RequestContext $requestContext = null)
+    {
+        $this->localDomain = $localDomain;
+        $this->requestContext = $requestContext;
+    }
+
+    public function configureLocalDomain(\Swift_Transport_AbstractSmtpTransport $transport)
```

Comment:
What about renaming if to just `configure`, so that we can use this for other things in the future if needed?